src/py_obs/osc.py

```diff
@@ -49,14 +53,41 @@ async def api_request(
         except aiohttp.ClientResponseError as cre_exc:
             raise ObsException(**cre_exc.__dict__) from cre_exc
 
+    async def get_auth_header(self) -> hash:
+        async with aiohttp.ClientSession() as session:
+            async with session.get(self.ibs_url_about) as r:
+                challenge_parameters = {
+                    'realm': r.headers['WWW-Authenticate'].split(',')[0].split('=')[-1].replace('"',""),
+                    'payload': r.headers['WWW-Authenticate'].split(',')[1].split('=')[-1].replace('"',"")
+                }
+                CREATED_TIMESTAMP = str(int(time.time()))
+                SIGNATURE_STRING = f"(created): {CREATED_TIMESTAMP}"
+                cmd = ['ssh-keygen', '-Y', 'sign', '-f', self.private_key_path, '-q', '-n',
+                        challenge_parameters['realm']]
+                s = subprocess.run(cmd, input=SIGNATURE_STRING, text=True, capture_output=True)
+
+                signature = ''.join(s.stdout.split('\n')[1:-2])
+                auth_header = { 'Authorization': f'Signature keyId="{self.username}",algorithm="ssh", signature="{signature}",headers="(created)",created="{CREATED_TIMESTAMP}"'}
+            async with session.get(self.ibs_url_about, headers=auth_header) as r:
+                assert str(r.status) == '200'
+                return auth_header
+
     def __post_init__(self) -> None:
-        self._session = aiohttp.ClientSession(
-            auth=aiohttp.BasicAuth(login=self.username, password=self.password),
-            raise_for_status=True,
-            base_url=self.api_url,
-            # https://github.com/openSUSE/open-build-service/issues/13737
-            headers={"Accept": "application/xml; charset=utf-8"},
-        )
+        if not re.match('.*suse\.de', self.api_url):
+            self._session = aiohttp.ClientSession(
+                auth=aiohttp.BasicAuth(login=self.username, password=self.password),
+                raise_for_status=True,
+                base_url=self.api_url,
+                # https://github.com/openSUSE/open-build-service/issues/13737
+                headers={"Accept": "application/xml; charset=utf-8"},
+            )
+        else:
+            auth_header = self.get_auth_header()
```

Comment:
This will not work. `get_auth_header` will return a promise, but you actually want the dictionary here. This whole function must be `async`, which is **not** possible with a dataclass' `__post_init__()`. So the initialization must be moved into an async classmethod.
```suggestion
            auth_header = await self.get_auth_header()
```

Reply:
I've never used await/async actually, but from what I read async method will return an Awaitable which must be awaited once. Fine with that but then how can this be handled if __post_init cannot have await statement ?

Reply:
Create a new classmethod for this, e.g.:
```Python
async def create_osc(…params) -> Osc:
    # your code here
```

Reply:
Ok that's what I started to do then we'll have to call the initailization method explicitly